main.cpp

```diff
@@ -51,23 +51,438 @@ If you need inspiration for what to write, take a look at previously approved st
+starPlayer("Neymar")
+{
+    std::cout << "SoccerTeam constructor called" << std::endl;
+};
+SoccerTeam::~SoccerTeam()
+{
+    std::cout << "SoccerTeam destructor called" << std::endl;
+}
+
+void SoccerTeam::transferPlayerToTeam(std::string transferToTeam, std::string transferFromTeam)
+{
+    std::cout << "Transferring from team " << transferFromTeam << " to team " << transferToTeam << std::endl;
+
+}
+
+
+double SoccerTeam::calculateWinProbability(float stoppageTime, int numberOfGoalsCurrentGame)
+{
+    if((stoppageTime < 3.0f) && (numberOfGoalsCurrentGame < 2))
+    {
+        std::cout << "Win probability over 50%" << std::endl;
+        return 0.6;
+    }
+
+    else
+    {
+        std::cout << "Win probability less than 50%" << std::endl;
+        return 0.4;
+    }
+
+}
+
+
+int SoccerTeam::calculateTotalSeasonGoals(int numberOfGoalsCurrentGame, int numberOfGoalsUntilNow)
+{
+    return (numberOfGoalsCurrentGame + numberOfGoalsUntilNow);
+}
 /*
  UDT 3:
  */
+struct DogShow
+{
+    std::string breedName = "Pitbull";
+    int weightInPounds;
+    std::string furColor;
+    float entranceFee;
+    float totalWinnings;
+    
+    float calculateCostToEnterDogShow(int weightInPounds = 30, std::string furColor = "brown", float entranceFee = 1000.54f);
+    void predictWinProbabilityDogShow(std::string breedName, int weightInPounds, std::string furColor);
+    float calculateTotalEarningsDogShow(float totalWinnings, float entranceFee = 2000.0f); 
+
+    DogShow();
+    ~DogShow();
+
+
+    struct Animal
+    {
+        int numberOfLegs = 4;
+        int theoreticalSpeed;
+        std::string animalType;
+        std::string countryOfShow;
+        int ageOfAnimal;
+
+        bool determineAnimalShow(std::string animalType, std::string countryOfShow = "USA");
+        int calculateMaxLandSpeed(int theoreticalSpeed = 40);
+        bool determineAnimalHealth(std::string animalType = "dog", int ageOfAnimal = 13);
+
+        Animal();
+        ~Animal();
+    };
+
+    
+
+};
+
+DogShow::DogShow():
+weightInPounds(135),
+furColor("black"),
+entranceFee(1500.0f),
+totalWinnings(4500.0f)
+{
+    std::cout << "Dog constructor called" << std::endl;
+}
+
+DogShow::~DogShow()
+{
+    std::cout << "Dog destructor called" << std::endl;
+}
+
+float DogShow::calculateCostToEnterDogShow(int weightInPounds, std::string furColor, float entranceFee)
+{
+    int totalCost = entranceFee;
+    if(weightInPounds > 50)
+    {
+        totalCost += 50.45;
+    }
+    else if(furColor == "white")
+    {
+        totalCost+= 75.63;
+    }
+
+    else
+    {
+        std::cout << "No additional costs" << std::endl;
+    }
+
+    return totalCost;
+
+}
+
+
+void DogShow::predictWinProbabilityDogShow(std::string breedName, int weightInPounds, std::string furColor)
+{
+    float winProbability  = 0.0f;
+
+    std::string breeds[5] = {"Pitbull", "German Shepherd", "Lab", "Great Dane", "Mastiff"};
+
+    for(int i = 0; i < 5; i++)
+    {
+        if(breedName == breeds[i])
```

Comment:
Everywhere that you have `i++`, `i--`, `i -= 1;` or `i += 1;`, you should use `pre-increment/decrement`. 
You are currently doing `post-increment/decrement`.  
you can think of post- and pre- increment like this:
```
int preIncrement(int& i)
{
    i += 1;
    return i; //both the returned value AND i have changed
}
​
int postIncrement(int& i)
{
    int oldVal = i;
    i += 1;
    return oldVal; //i has changed but the value being returned does not reflect this change.
}
```
more information can be found here explaining why `pre-increment/decrement` is preferred over `post-increment/decrement`: https://en.cppreference.com/w/cpp/language/operator_incdec
- [ ] revise